ui/article-aem-renderer/src/main/java/org/cru/godtools/article/aem/service/AemArticleManager.kt

```diff
@@ -156,9 +156,11 @@ class AemArticleManager @VisibleForTesting internal constructor(
     }
 
     init {
-        aemDb.aemImportDao().all
-            .filter { it.isStale() }
-            .forEach { coroutineScope.launch { syncAemImport(it.uri, false) } }
+        coroutineScope.launch {
+            aemDb.aemImportDao().getAll()
+                .filter { it.isStale() }
+                .forEach { launch { syncAemImport(it.uri, false) } }
+        }
```

Comment:
So before only the ```syncAemImport``` was wrapped in coroutine scope, but now all of this is. Was there a chance before it was wrapped in coroutine scope, that it would try to run ```syncAemImport``` before ```getAll``` finished?

Reply:
nope, `getAll()` would have to return before either the `filter` statement or the `syncAemImport` method even started executing.

The entire block is wrapped in a `launch` because it needs to process on a background thread, then I launch separate coroutines for each found stale `AemImport` so that they can execute in parallel on background threads.

Reply:
Gotcha, thanks!